test/index.js

```diff
@@ -522,6 +537,86 @@ test('equals', function() {
   eq(Z.equals(Just(0).constructor, Maybe), true);
 });
 
+test('lte', function() {
+  eq(Z.lte.length, 2);
+  eq(Z.lte.name, 'lte');
+
+  eq(Z.lte(null, null), true);
+  eq(Z.lte(null, undefined), false);
+  eq(Z.lte(undefined, null), false);
+  eq(Z.lte(undefined, undefined), true);
+  eq(Z.lte(false, false), true);
+  eq(Z.lte(false, true), true);
+  eq(Z.lte(true, false), false);
+  eq(Z.lte(true, true), true);
+  eq(Z.lte(new Boolean(false), new Boolean(false)), true);
+  eq(Z.lte(new Boolean(false), new Boolean(true)), true);
+  eq(Z.lte(new Boolean(true), new Boolean(false)), false);
+  eq(Z.lte(new Boolean(true), new Boolean(true)), true);
+  eq(Z.lte(false, new Boolean(false)), false);
+  eq(Z.lte(new Boolean(false), false), false);
+  eq(Z.lte(true, new Boolean(true)), false);
+  eq(Z.lte(new Boolean(true), true), false);
+  eq(Z.lte(0, 0), true);
+  eq(Z.lte(0, -0), true);
+  eq(Z.lte(-0, 0), true);
+  eq(Z.lte(-0, -0), true);
+  eq(Z.lte(NaN, NaN), true);
+  eq(Z.lte(Infinity, Infinity), true);
+  eq(Z.lte(Infinity, -Infinity), false);
+  eq(Z.lte(-Infinity, Infinity), true);
+  eq(Z.lte(-Infinity, -Infinity), true);
+  eq(Z.lte(NaN, Math.PI), false);
+  eq(Z.lte(Math.PI, NaN), false);
+  eq(Z.lte(new Number(0), new Number(0)), true);
+  eq(Z.lte(new Number(0), new Number(-0)), true);
```

Comment:
Preserving referential transparency seems more important to me than having a single identity. If `x` is equal to `y`, I don't want `1 / x` to give `Infinity` while `1 / y` gives `-Infinity`. One could argue, though, that division be zero is nonsensical. If you'd like to take that position, @i-am-tom, I'm open to considering the possibility of treating `0` and `-0` as equal.